Feels like something ain't right here...

Before it was printing the error

```
CMake Error in CMakeLists.txt:                                                                     
  Target "hello" INTERFACE_INCLUDE_DIRECTORIES property contains path:                             
                                                                                                   
    "/home/bocchi/hello_fetchcontent/include"                                                      
                                                                                                   
  which is prefixed in the source directory.                                                       
```

But everything actually still built and ran just fine, but ChatGPT told me to change the `target_include_directories` to this:

```
target_include_directories(hello PUBLIC                                                            
    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>                                         
    $<INSTALL_INTERFACE:include>                                                                   
)      
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,15 +1,14 @@
 cmake_minimum_required(VERSION 3.14)
 project(hello_fetchcontent VERSION 1.0)
 
-add_library(hello src/hello.cpp)
+add_library(hello SHARED src/hello.cpp)
+add_library(hello_fetchcontent::hello ALIAS hello)
 
 target_include_directories(hello PUBLIC 
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    $<INSTALL_INTERFACE:include>
-)
+    $<INSTALL_INTERFACE:include>)
 
 
-add_library(hello_fetchcontent::hello ALIAS hello)
 
 install(TARGETS hello
         EXPORT helloTargets

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,31 +1,8 @@
-# cmake_minimum_required(VERSION 3.11)
-# project(hello_fetchcontent VERSION 1.0)
-# 
-# add_library(hello_fetchcontent src/hello.cpp)
-# add_library(hello_fetchcontent::hello_fetchcontent ALIAS hello_fetchcontent)
-# 
-# target_include_directories(hello_fetchcontent PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
-# 
-# install(TARGETS hello_fetchcontent
-#         EXPORT HelloFetchContentTargets
-#         LIBRARY DESTINATION lib
-#         ARCHIVE DESTINATION lib
-#         RUNTIME DESTINATION bin)
-# 
-# install(DIRECTORY include/ DESTINATION include)
-# 
-# install(EXPORT HelloFetchContentTargets
-#         FILE HelloFetchContentTargets.cmake
-#         NAMESPACE hello_fetchcontent::
-#         DESTINATION lib/cmake/hello_fetchcontent)
-
 cmake_minimum_required(VERSION 3.14)
 project(hello_fetchcontent VERSION 1.0)
 
 add_library(hello src/hello.cpp)
 
-# target_include_directories(hello PUBLIC include)
-# Specify include directories for when building the library
 target_include_directories(hello PUBLIC 
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>

--- a/README.md
+++ b/README.md
@@ -15,3 +15,12 @@ MyLibrary/
 └── CMakeLists.txt
 ```
 
+> Include directories usage requirements commonly differ between the build-tree and the install-tree. The BUILD_INTERFACE and INSTALL_INTERFACE generator expressions can be used to describe separate usage requirements based on the usage location. Relative paths are allowed within the INSTALL_INTERFACE expression and are interpreted as relative to the installation prefix. Relative paths should not be used in BUILD_INTERFACE expressions because they will not be converted to absolute. For example:
+
+```
+target_include_directories(mylib PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/mylib>
+  $<INSTALL_INTERFACE:include/mylib>  # <prefix>/include/mylib
+)
+```
+

